File: packages/sdk-ruby/lib/bidstack_crm/resources/activities.rb
require_relative "base"

module BidstackCrm
  module Resources
    class Activities < Base
      def list(page: 1, per_page: 50, entity_type: nil, entity_id: nil)
        params = { page: page, perPage: per_page }
        params[:entityType] = entity_type if entity_type
        params[:entityId]   = entity_id   if entity_id
        get("/activities", params: params)
      end

      def find(activity_id) = get("/activities/#{activity_id}")
      def create(data, idempotency_key: nil) = post("/activities", body: data, idempotency_key: idempotency_key)
      def update(activity_id, data) = patch("/activities/#{activity_id}", body: data)
      def destroy(activity_id) = delete("/activities/#{activity_id}")
    end
  end
end

File: packages/sdk-python/bidstack_crm/resources/activities.py
"""Activities resource."""
from __future__ import annotations
from typing import Any
from ._base import BaseResource


class ActivitiesResource(BaseResource):
    def list(self, *, entity_type: str | None = None, entity_id: str | None = None, page: int = 1, per_page: int = 50) -> dict[str, Any]:
        params: dict[str, Any] = {"page": page, "perPage": per_page}
        if entity_type:
            params["entityType"] = entity_type
        if entity_id:
            params["entityId"] = entity_id
        return self._get("/activities", params=params)

    def get(self, activity_id: str) -> dict[str, Any]:
        return self._get(f"/activities/{activity_id}")

    def create(self, data: dict[str, Any], idempotency_key: str | None = None) -> dict[str, Any]:
        return self._post("/activities", json=data, idempotency_key=idempotency_key)

    def update(self, activity_id: str, data: dict[str, Any]) -> dict[str, Any]:
        return self._patch(f"/activities/{activity_id}", json=data)

    def delete(self, activity_id: str) -> None:
        self._delete(f"/activities/{activity_id}")
